lib/addons/helm-addon/kubectl-provider.ts

```diff
@@ -48,6 +48,15 @@ export interface HelmChartConfiguration {
      * Optional values for the helm chart. 
      */
     values?: Values
+
+    /**
+     * Indicate the helm chart provided uses dependency mode (https://helm.sh/docs/helm/helm_dependency/).
+     * Dependency mode is widely used in `aws-samples/ssp-eks-add-ons` repository, for example:
+     * https://github.com/aws-samples/ssp-eks-add-ons/blob/main/add-ons/appmesh-controller/Chart.yaml
+     * Dependency mode requires the chart values to be wrapped within the chart name.
+     * This value is only used when AddOns are deployed via GitOps way: `ArgoGitOpsFactory.enableGitOps()`
+     */
+    dependencyMode?: boolean
```

Comment:
Let's specify that the default is `true` and this field is only used to `turn off dependency mode in case customers choose to copy the whole helm chart into their repo`

Reply:
doc updated as suggested 👍 

the default is set to `true`, not inside the interface, but in `lib/addons/helm-addon/index.ts`:

`const dependencyMode = this.props.dependencyMode ?? true;`

Reply:
I meant in jsdoc using `@default`:

```typescript
/**
* @default true
*/
```

We use it here: https://aws-quickstart.github.io/cdk-eks-blueprints/api/
Also vscode support is great for such cases.

Reply:
Thanks - added.